### ISSUE TITLE
Add support for Django 1.7

### ISSUE THREAD

**Issue by apendleton** [chop-dbhi/avocado#311]
I'm starting on a project for which I'd like to use avocado (and the rest of the Harvest family), but I have another dependency that hard-depends on Django 1.7, so I'm trying to see if I can get avocado working against it. I have a working branch here:

https://github.com/apendleton/avocado/tree/django1.7

And I think I've gotten the low-hanging fruit either resolved or hacked around (mostly issues with South not supporting 1.7), but the remaining error and three failures seem to involve some pretty complex behavior in the DataQuery and/or search internals, and it's not obvious to me what's going wrong. Do you guys have any pointers?

Once I get these hammered out I can submit a PR.

**Comment by bruth** [comment]
This is great! Thank you for taking the time to put this together. I checked out the code and looked into the issues. There were a few existing bugs in the tests, here is the patch you can apply to your branch.

```diff
diff --git a/tests/cases/models/tests.py b/tests/cases/models/tests.py
index 4f6472b..991d02b 100644
--- a/tests/cases/models/tests.py
+++ b/tests/cases/models/tests.py
@@ -506,8 +506,8 @@ class DataQueryTestCase(TestCase):
         self.assertEqual(query.json, json)
 
     def test_count(self):
-        salary_field = DataField.objects.get_by_natural_key(
-            'tests', 'title', 'salary')
+        c = DataConcept.objects.get(fields__model_name='title',
+                                    fields__field_name='salary')
 
         json = {
             'context': {
@@ -516,7 +516,7 @@ class DataQueryTestCase(TestCase):
                 'value': '1000'
             },
             'view': [{
-                'concept': salary_field.id,
+                'concept': c.id,
                 'visible': True,
             }]
         }
@@ -633,7 +633,7 @@ class DataQueryTestCase(TestCase):
             '"tests_office"."id") WHERE "tests_title"."boss" = True '
             .replace(' ', ''))
 
-        query = DataQuery({'view': {'ordering': [(1, 'desc')]}})
+        query = DataQuery({'view': {'ordering': [(c.pk, 'desc')]}})
         queryset = Employee.objects.all().distinct()
         self.assertEqual(
             unicode(query.apply(queryset=queryset).query).replace(' ', ''),
diff --git a/tests/cases/search/tests.py b/tests/cases/search/tests.py
index 1d7f4da..2dbe212 100644
--- a/tests/cases/search/tests.py
+++ b/tests/cases/search/tests.py
@@ -160,11 +160,17 @@ class FieldSearchTest(SearchTest):
 
     def test_data(self):
         "Test search via the data itself."
+        last_name = DataField.objects.get(model_name='employee',
+                                          field_name='last_name')
+        first_name = DataField.objects.get(model_name='employee',
+                                           field_name='first_name')
+        title_name = DataField.objects.get(model_name='title',
+                                           field_name='name')
         values = [
-            ('Jones', [6]),
-            ('Programmer', [2]),
-            ('Erick', [5]),
-            ('CEO', [2]),
+            ('Jones', [last_name.pk]),
+            ('Programmer', [title_name.pk]),
+            ('Erick', [first_name.pk]),
+            ('CEO', [title_name.pk]),
         ]
 
         for v, ids in values:
@@ -229,12 +235,17 @@ class ConceptSearchTest(SearchTest):
 
     def test_data(self):
         "Test search via the data itself."
-
+        last_name = DataConcept.objects.get(fields__model_name='employee',
+                                            fields__field_name='last_name')
+        first_name = DataConcept.objects.get(fields__model_name='employee',
+                                             fields__field_name='first_name')
+        title_name = DataConcept.objects.get(fields__model_name='title',
+                                             fields__field_name='name')
         values = [
-            ('Jones', [6]),
-            ('Programmer', [2]),
-            ('Erick', [5]),
-            ('CEO', [2]),
+            ('Jones', [last_name.pk]),
+            ('Programmer', [title_name.pk]),
+            ('Erick', [first_name.pk]),
+            ('CEO', [title_name.pk]),
         ]
 
         for v, ids in values:
```

One requirement before submitting the pull request is to sign off on all the commits noted in our [contributing guide](https://github.com/chop-dbhi/avocado/blob/master/CONTRIBUTING.md#contributing).

**Comment by apendleton** [comment]
Awesome, thanks, that worked.

**Comment by bruth** [comment]
Fixed in #312. Thanks again for the contribution!
